chains/evm/contracts/onRamp/OnRamp.sol

```diff
@@ -463,9 +483,9 @@ contract OnRamp is IEVM2AnyOnRampClient, ITypeAndVersion, Ownable2StepMsgSender
     if (extraArgs.length >= 4 && bytes4(extraArgs[:4]) == ExtraArgsCodec.GENERIC_EXTRA_ARGS_V3_TAG) {
       resolvedArgs = ExtraArgsCodec._decodeGenericExtraArgsV3(extraArgs);
 
-      if (resolvedArgs.tokenReceiver.length != 0) {
-        this.validateDestChainAddress(resolvedArgs.tokenReceiver, destChainConfig.addressBytesLength);
-      }
+      resolvedArgs.tokenReceiver = resolvedArgs.tokenReceiver.length != 0
+        ? _validateDestChainAddress(resolvedArgs.tokenReceiver, destChainConfig.addressBytesLength)
+        : resolvedArgs.tokenReceiver;
```

Comment:
Why not use the original if statement? This effectively sets resolvedArgs.tokenReceiver to resolvedArgs.tokenReceiver if zero length. The same behaviour but more expensive and harder to read right?

Reply:
Original if statement did not assign tokenReceiver to the validated and trimmed token receiver. But yeah I think you and Kyle are right using an if statement would be more readable and gas efficient

Reply:
Fixed [efe68dc](https://github.com/smartcontractkit/chainlink-ccip/pull/1476/commits/efe68dca2d4bf48f572b73cc81a745b1a4f33b06)